### ISSUE TITLE
Clean up QEMU documentation for v0.1.0 release

### ISSUE THREAD

**Issue by macpijan** [Dasharo/dasharo-issues#536]
**Dasharo version (if applicable)**

v0.1.0

**Dasharo variant (if applicable)**

Emulation QEMU Q35

**Affected component(s) or functionality (if applicable)**

Documentation

**Brief summary**

- [x] Review and update [Test Matrix](https://docs.dasharo.com/variants/qemu_q35/test-matrix/)
  - do we really test for Windows booting, for example?
  - how do we test Secure Boot? Can we boot some OS with existing documentation? 
- [ ] Consider providing a script, instead of repeating the same long QEMU run command in the documentation multiple times
- [x] #541

**Additional context**

**Comment by pietrushnic** [comment]
@macpijan 

>  Consider providing a script, instead of repeating the same long QEMU run command in the documentation multiple times

I'm trying to understand which commands have to be repeated. Every reasonable shell had a copy-pastable history command, no matter the length. We can provide a wrapper if you think it helps anyone, but I have doubts about that as we are talking in the context of Dasharo Universe documentation. Providing a script means we have to do more than just copying because we need a repository with the script. We have to consider who is the user of the QEMU Q35 target. From my perspective, there can be the following reasons to use Dasharo (UEFI) for QEMU Q35:
- Dasharo validation - in such cases, there are already [scripts](https://github.com/Dasharo/open-source-firmware-validation/pull/52/files#diff-69e6355529b2609a8ddcccd04f8cd299d0fe65c19c246c853a31b9eb96fe6818) to download some version of the release and use it directly in OSFV
- Dasharo development - in this case, the user will typically go through the building manual and has quite good QEMU-fu to deal with all required parameters. We already have some [dev docs](https://docs.dasharo.com/variants/qemu_q35/development/)
- Production integration  - one KPI or measure of success would be to introduce our OVMF version to XCP-ng, Proxmox or other projects that manage VMs or create some footprint in Confidential VMs, but that is the long-term vision of this build

Are there any use cases I'm missing?

To sum up:
- Dasharo validation - we need initial deployment, which describes how to use files from release as well as improve OSFV scripts to handle certain releases.
- Dasharo dev - we don't have to do anything IMHO because we have dev docs.
- Production - we don't have such use cases, and if we do, we will rely on OS deployment mechanisms.

**Comment by pietrushnic** [comment]
@macpijan ping

**Comment by macpijan** [comment]
@pietrushnic 

> I'm trying to understand which commands have to be repeated. 

My concern was the QEMU command like we have here: https://docs.dasharo.com/variants/qemu_q35/development/

It is repeated 5 or more times in the qemu documentation. Do we envision some improvements in the arguments? We already introduced some improvements to the qemu commandline when creating new tests and feature in the last couple of weeks. 

Can we assure we will keep all of them in sync and modify in all of these places?

**Comment by macpijan** [comment]
The v0.1.0 has been released with some improvements: https://docs.dasharo.com/variants/qemu_q35/releases/
I will move it to another milestone so we can still consider these further improvements discussed here.

**Comment by pietrushnic** [comment]
> My concern was the QEMU command like we have here: https://docs.dasharo.com/variants/qemu_q35/development/
> It is repeated 5 or more times in the qemu documentation.

I still don't get it, do you mean this command:

```
qemu-system-x86_64 -machine q35,smm=on \
    -global driver=cfi.pflash01,property=secure,value=on \
    -drive if=pflash,format=raw,unit=0,file=Build/OvmfX64/DEBUG_GCC5/FV/OVMF_CODE.fd,readonly=on \
    -drive if=pflash,format=raw,unit=1,file=Build/OvmfX64/DEBUG_GCC5/FV/OVMF_VARS.fd \
    -debugcon file:debug.log -global isa-debugcon.iobase=0x402 \
    -global ICH9-LPC.disable_s3=1
```

? Where it is repeated five times? Can you link?

What I was able to find;
- [dev intro](https://docs.dasharo.com/variants/qemu_q35/development/#intro)
- [recovery x2](https://docs.dasharo.com/variants/qemu_q35/recovery/)

> Do we envision some improvements in the arguments?

I'm always okay with justified improvements, but I still don't understand repeating the part you mentioned and your POV in case of improvements. Since you improved the approach to the qemu command line, then, I have nothing to say except potential issues with my use case, which I already explained when we planned open-sourcing OSFV.

> Can we assure we will keep all of them in sync and modify in all of these places?

I read it as a rhetorical question, but I will answer anyway. Of course, we will not be able to keep that in sync, which was proven many times.

I'm not against wrapping the scripts. IMHO, you have unexpressed assumptions about the users of such script. Are you considering another user than `Dasharo validation` described above?

> If no script, then maybe at least use it once in documentation and reference from other places? Or reference a script from OSFV repo?

I'm ok with both, but the second approach could be better and more scalable. Is simple `wget/curl` ok in such situations? That means that the script call would not change, but let's make this assumption. I need information.

**Comment by macpijan** [comment]
> I still don't get it, do you mean this command:

I meant that one, and we've made some improvements since then.

> I'm ok with both, but the second approach could be better and more scalable. Is simple wget/curl ok in such situations? That means that the script call would not change, but let's make this assumption. I need information.

The script for running Dasharo qemu will evolve, I am sure of it.
Now that I think of it, maybe it should be developed along with the QEMU fw release (so the edk2 repo), and be releases as one of the artifacts? That way we always have script matching given release, and whoever needs to use it, can wget/curl from matching tag.

**Comment by pietrushnic** [comment]
> Now that I think of it, maybe it should be developed along with the QEMU fw release (so the edk2 repo), and be releases as one of the artifacts? That way we always have script matching given release, and whoever needs to use it, can wget/curl from matching tag.

Definitely! This is the correct approach; we should keep the script as close to the code as possible, and it looks like the qemu running Dasharo (OVMF) build is closer to our edk2 fork than to OSFV. What we should target is script parameters compatible with what Dasharo (OVMF) can do instead of what OSVF can do.

**Comment by krystian-hebel** [comment]
Among other changes, I've made that command significantly shorter in https://github.com/Dasharo/docs/pull/825, and linked to one source of truth instead of repeating it everywhere. While some of the removed options could be used for mid-level (lower than normal output, higher than QEMU monitor) debugging, they aren't required for normal use. Disabling S3 is also not required for coreboot+UEFI, at least not for the reason OVMF required it to be disabled.

Let me know if you still think it is worth to add a script for starting QEMU.

**Comment by krystian-hebel** [comment]
Do we have plans for updating tests description on docs.dasharo.com? I'm having difficulties filling up test matrix, some tests are different in OSFV scripts than described on that page.

**Comment by BeataZdunczyk** [comment]
Closing this issue and milestone as the v0.2.0 release is published.  Release notes are here: https://docs.dasharo.com/variants/qemu_q35/releases/#v010-2023-12-06

> Review and update [Test Matrix](https://docs.dasharo.com/variants/qemu_q35/test-matrix/)

updated test matrix: https://docs.dasharo.com/variants/qemu_q35/test-matrix/

 > Consider providing a script, instead of repeating the same long QEMU run command in the documentation multiple times
 
Simplified command here: https://docs.dasharo.com/variants/qemu_q35/initial-deployment/

**Comment by BeataZdunczyk** [comment]
> Do we have plans for updating tests description on docs.dasharo.com? I'm having difficulties filling up test matrix, some tests are different in OSFV scripts than described on that page.

Feel free to raise an issue specifying the discrepancies @krystian-hebel
